src/sinks/util/buffer/mod.rs

```diff
@@ -9,9 +9,11 @@ pub mod partition;
 
 pub use partition::{Partition, PartitionBuffer, PartitionInnerBuffer};
 
-#[derive(Serialize, Deserialize, Debug, Copy, Clone, Eq, PartialEq)]
-#[serde(rename_all = "lowercase")]
+#[derive(Serialize, Deserialize, Debug, Derivative, Copy, Clone, Eq, PartialEq)]
+#[derivative(Default)]
+#[serde(rename_all = "snake_case")]
```

Comment:
Do we really want `snake_case` here? That seems to unnecessarily make this a breaking change.

Reply:
`snake_case` used in `gcp_cloud_storage` sink: https://github.com/timberio/vector/blob/6281932469d9bcef34edaf672b52edc804d49732/src/sinks/gcp/cloud_storage.rs#L144
Should both be `lowercase`?

Reply:
Snake case is used in the GCP sinks because we are encoding to a GCP REST API that requires the use of snake case. They use their own enums though. This is used by other sinks that don't have the same requirements.